lib/graphql/batch/loader.rb

```diff
@@ -54,13 +54,16 @@ def resolve #:nodoc:
     # For Promise#sync
     def wait #:nodoc:
       if executor
-        executor.loaders.delete(loader_key)
         executor.resolve(self)
       else
         resolve
       end
     end
 
+    def resolved?
+      queue.empty?
```

Comment:
Use `@queue.nil? || @queue.empty?` to avoid unnecessarily creating an empty array.  You can also use the same method in `resolve` to `return if resolved?` before `load_keys = queue` instead of using `return if load_keys.empty?` afterwards.